plugins/premium/powerpaste.md

```diff
@@ -95,47 +95,86 @@ tinymce.init({
 
 This option controls how content pasted from Microsoft Word is filtered.
 
-**Type:** `String`
+**Type:** `String` or `Function`
 
 **Default value:** `prompt`
 
-The supported values are:
-
-* `clean` - Preserve the structure of the content such as headings, tables and lists but remove inline styles and classes. This results in simple content that uses the site's CSS stylesheet while retaining the semantic structure from the original document.
-* `merge` - Preserve the inline formatting and structure of the original document. Invalid and proprietary styles, tags and attributes are still removed ensuring that the HTML is valid while more closely matching the original document formatting.
-* `prompt` - Prompt the user to choose between the clean and merge options after attempting to paste Microsoft Word content.
+{% include plugins/powerpaste_import_types.md %}
 
 > **Note:** When using the Windows operating system, copying and pasting content from Microsoft Word 2013 (or later) in "Protected View" will result in plain, unformatted text. This is due to how Protected View interacts with the clipboard.
 
+#### Example: `powerpaste_word_import` using an asynchronous function
+
+```js
+tinymce.init({
+  selector: 'textarea',  // change this value according to your HTML
+  plugins: 'powerpaste',
+  powerpaste_word_import: function() {
+    // use a native confirm dialog to prompt the user to choose between clean and merge
+    if (confirm('Would you like to keep formatting?')) {
+      return Promise.resolve('merge');
+    } else {
+      return Promise.resolve('clean');
+    }
```

Comment:
```suggestion
    return new Promise(function (resolve) {
      if (confirm('Would you like to keep formatting?')) {
        resolve('merge');
      } else {
        resolve('clean');
      }
    });
```

Because `confirm` is a synchronous function, there isn't any functional difference between these two. However, I feel like this form would be easier to adapt to using a real dialog. Feel free to ignore